cmd/proxy/main.go

```diff
@@ -0,0 +1,111 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"log/slog"
+	"os"
+
+	"github.com/absmach/propeller/proxy"
+	"github.com/caarlos0/env/v11"
+	"golang.org/x/sync/errgroup"
+)
+
+const svcName = "proxy"
+
+type config struct {
+	LogLevel string `env:"PROXY_LOG_LEVEL"           envDefault:"info"`
+
+	BrokerURL  string `env:"PROXY_MQTT_ADDRESS"          envDefault:"tcp://localhost:1883"`
+	PropletKey string `env:"PROXY_PROPLET_KEY,notEmpty"`
+	PropletID  string `env:"PROXY_PROPLET_ID,notEmpty" `
+	ChannelID  string `env:"PROXY_CHANNEL_ID,notEmpty"`
+
+	ChunkSize    int    `env:"PROXY_CHUNK_SIZE"             envDefault:"512000"`
+	Authenticate bool   `env:"PROXY_AUTHENTICATE"           envDefault:"false"`
+	Token        string `env:"PROXY_REGISTRY_TOKEN"         envDefault:""`
```

Comment:
Do we need token and password or it is either one?

Reply:
Its either of the 2